src/sage/rings/function_field/function_field_polymod.py

```diff
@@ -34,12 +34,16 @@
 from sage.rings.integer import Integer
 from sage.categories.homset import Hom
 from sage.categories.function_fields import FunctionFields
+from sage.categories.number_fields import NumberFields
 
 from .element import FunctionFieldElement
 from .element_polymod import FunctionFieldElement_polymod
 from .function_field import FunctionField
 from .function_field_rational import RationalFunctionField
 
+_FunctionFields = FunctionFields()
+_NumberFields = NumberFields()
```

Comment:
Why do we pin these in memory here? Is the `FunctionField_polymod` called so often in tight loops that the extra construction of the category has a noticeable impact?

`NumberFields` should not be because the genus shouldn't be called that often (we might even be okay just doing the import there as well, but that isn't so important).

Reply:
I wanted to save some micro seconds. But you are right. It is not likely to be called in tight loops. 

OK. Done.